onyo/lib/items.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import onyo.lib.onyo
+import onyo.lib.inventory
+import onyo.lib.pseudokeys
+from onyo.lib.utils import DotNotationWrapper
+
+
+if TYPE_CHECKING:
+    from typing import (
+        Any,
+        Mapping,
+        TypeVar,
+    )
+
+    _KT = TypeVar("_KT")  # Key type.
+    _VT = TypeVar("_VT")  # Value type.
+
+
+def resolve_alias(key: Any) -> Any:
+    """Return the target of a key alias."""
+    try:
+        return onyo.lib.pseudokeys.PSEUDOKEY_ALIASES[key]
+    except KeyError:
+        return key
+
+
+class Item(DotNotationWrapper):
+    """An Item an Inventory can potentially track.
```

Comment:
Interesting word "potentially".

Reply:
:-D

Don't have a good phrasing yet. Point being, that this thing may be the result of `new --keys ..`, piped in YAML, merged templates etc. that are a state in memory that isn't any committed thing yet.